### PR TITLE
Add a resolvable JavaScript entry point to base-styles

### LINKED ISSUE #39239: Base Styles: npm package does not have a resolvable entry point
### Description

The npm package `@wordpress/base-styles` does not contain an entry point that's resolvable via JavaScript, which causes certain build tools to fail when using this package outside of the Gutenberg project. As npm is a JavaScript registry, the package should be resolvable with JavaScript.

This package originally had an `index.js` file, but it was removed in [this commit](https://github.com/WordPress/gutenberg/commit/988d3a0edf9f8eca36e4003b241d050f9a8df6ea).

### Step-by-step reproduction instructions

1. There is no `index.js` file in `./packages/base-styles/`.

### Screenshots, screen recording, code snippet

_No response_

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by joshuafredrickson**
## Description

Adds a resolvable JavaScript entry point. Fixes #39239.

## Checklist:
- [X] My code is tested.
- [X] My code follows the WordPress code style. 
- [X] My code follows the accessibility standards. 
- [X] I've tested my changes with keyboard and screen readers. 
- [X] My code has proper inline documentation. 
- [X] I've included developer documentation if appropriate. 
- [X] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [X] I've updated related schemas if appropriate.

### CODE REVIEW

**Review by gziolo — APPROVED** [review]
I'm not quite sure what type of errors it generates, but it won't harm anything to include an empty entry point 👍🏻

File: packages/base-styles/index.js [hunk 1]
```diff
@@ -0,0 +1 @@
+// index.js
```
Comment (gziolo) [inline packages/base-styles/index.js]: One more thing before we land. Can we update the comment to be more informative to ensure it isn't deleted in the future?

Comment (joshuafredrickson) [reply packages/base-styles/index.js]: Coming right up!

**Comment by gziolo** [comment]
Looks great, thank you!

**Pull request merged**